### ISSUE TITLE
diagnosis of unnecessary if gives false positives

### ISSUE THREAD

**Issue by MillhioreBT** [EmmyLuaLs/emmylua-analyzer-rust#472]
![Image](https://github.com/user-attachments/assets/9733027c-1582-4a2b-bb45-914b952de630)

![Image](https://github.com/user-attachments/assets/dd6b3192-5a1c-44c1-b8fe-6a33e42de5c6)

```lua
---@param color integer
---@param level integer
function Game.setWorldLight(color, level)
    local previousColor = worldLightColor
    local previousLevel = worldLightLevel

    worldLightColor = color
    worldLightLevel = level

    if worldLightColor ~= previousColor or worldLightLevel ~= previousLevel then
    -- Do something...
    end
end
```

**Comment by xuhuanzy** [comment]
@ribru17 I think we need to lower the display level of this diagnostic to `info` instead of `warning`.

**Comment by ribru17** [comment]
This seems more like a flow issue though, no? Ideally it would still display as `warning`, but correctly

**Comment by xuhuanzy** [comment]
> This seems more like a flow issue though, no? Ideally it would still display as `warning`, but correctly

Many people report that the inspection is incorrect or unnecessary because it's too strict, and lowering the display level is the best option.
